### ISSUE TITLE
Equally use singular for unit types as in the c# project

### ISSUE THREAD

**Issue by Shentoza** [haimkastner/unitsnet-js#29]
We're using the unitsnet-js as a way to have the same information in our frontend, that we also have in our backend.
However in this package all unitnames are in plural (e.g. `Meters`, `Millimeters`, `Inches` ) while in the c# project they are singular (e.g. `Meter`, `Millimeter`, `Inch` ) which makes conversion between our c# backend and our typescript frontend really painful.
Is this a conscious decision?

**Comment by haimkastner** [comment]
Hi @Shentoza 

Are you referring to the unit's enum? `LengthUnits` in the length unit case?

This enum convention is not the same as in the C#, and I do agree that it was better to be the same convention, but why is this so painful?

Can you please share a use case for why it's so matter?

I can think of a use case, when you want to pass the unit as part of an API response, for instance:
```json
{
 "value": 10,
 "unit": "Meters"
}
```

Then in the consumer, you want to do the following:
```typescript
const res = api_call()
const length = new Length(res.value, res.unit)
```

But, if this is the case, it doesn't matter what the enum key name is, but what the actual value is.

BTW, currently, the enums value of the units are random numbers (depending on the order, auto set number)

**Comment by Shentoza** [comment]
Because the object coming from my Backend is 
{ value: 10,
 unit: "Meter"}. I'm using the enum name here to have a more speaking response in case anyone not using UnitsNet is calling my API, where the enum value of e.g. 12 would be confusing.
 Now i have to do a conversion where I map the singular response of the backend to the plural naming of the typescript enum, s.t. I can create an `Length` object, (since there is only an `constructor(value: number, fromUnit?: LengthUnits);` constructor I need the actual enum value. And if I want to send the object back to the backend, I need to convert it back again to the singular form.
 
 
 I have seen that in the generator files ( `unit-generator.ts` ) in buildEnum it is specified to take the plural name, which should be easy to alter, no?

**Comment by haimkastner** [comment]
I understand, so basically if I will change the *value* of the enum from some random number to the unit non-plural  name

Like this one 
https://github.com/haimkastner/unitsnet-js/blob/master/src/length.g.ts#L4
```typescript
Meters = 'Meter',
``` 

It will work for you even if the enum key is still plural, like this:
```typescript
const length = new Length(res.value, res.unit as LengthUnits)
```

_I have seen that in the generator files ( unit-generator.ts ) in buildEnum it is specified to take the plural name, which should be easy to alter, no?_
Right, it's easy to change it, but it will break the package API (unless I will duplicate it and deprecate the old unit enumas I explain above).

**Comment by Shentoza** [comment]
> I understand, so basically if I will change the _value_ of the enum from some random number to the unit non-plural name
> 
> Like this one https://github.com/haimkastner/unitsnet-js/blob/master/src/length.g.ts#L4
> 
> ```ts
> Meters = 'Meter',
> ```
> 
> It will work for you even if the enum key is still plural, like this:
> 
> ```ts
> const length = new Length(res.value, res.unit as LengthUnits)
> ```
That would be great!

**Comment by haimkastner** [comment]
@Shentoza 
The new version `2.0.1` contains the fix from #30 published and ready in NPM.

**Comment by Shentoza** [comment]
> @Shentoza The new version `2.0.1` contains the fix from #30 published and ready in NPM.

Thank you so much. We're using yarn, is the package already pushed there as well? im taking a look into #31
